### PR TITLE
Feature donate button

### PR DESCRIPTION
**Pull request by jack-fdrv**
Show button only for product which was paired
Add extra product attribute for donated products
Upon order add small notice to order about donation

prepare:
- server config
- Add script with button. Hardcode url in theme editor.
- Fix problem with remix.
- Update cli

### CODE REVIEW

**Review by goliney — COMMENTED** [review]
Please provide instructions for getting the donation button to appear.

File: remix.config.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ if (
   process.env.SHOPIFY_APP_URL = process.env.HOST;
   delete process.env.HOST;
 }
-
+delete process.env.REMIX_DEV_ORIGIN;
```
[the remark this answers is not in the document]
Comment (jack-fdrv) [reply remix.config.js:14]: This fix issue with remix on run.

https://github.com/Shopify/cli/issues/2935

File: app/entry.server.jsx [hunk 2]
```diff
@@ -9,50 +9,55 @@ import { addDocumentResponseHeaders } from "./shopify.server";
...
-      }
-    );
-
-    setTimeout(abort, ABORT_DELAY);
-  });
+	addDocumentResponseHeaders(request, responseHeaders);
+	console.log(' ')
+	console.log('process.env.HOST:', process.env.SHOPIFY_APP_URL)
+	console.log('SHOPIFY_API_VERSION:', process.env.SHOPIFY_API_VERSION);
+	console.log(' ')
```
[the remark this answers is not in the document]
Comment (jack-fdrv) [reply app/entry.server.jsx:22]: There a lot of console log which we will need to remove before prod.

File: app/entry.server.jsx
[the remark this answers is not in the document]
Comment (jack-fdrv) [reply app/entry.server.jsx:1]: Yes, we added  getPair endpoint.

File: app/entry.server.jsx [hunk 3]
```diff
@@ -9,50 +9,55 @@ import { addDocumentResponseHeaders } from "./shopify.server";
...
-  request,
-  responseStatusCode,
-  responseHeaders,
-  remixContext,
-  _loadContext
+	request,
+	responseStatusCode,
+	responseHeaders,
+	remixContext,
+	_loadContext
```
[the remark this answers is not in the document]
Comment (jack-fdrv) [reply app/entry.server.jsx:16]: What if I used to use tabs? How we can make it works for both of us? =)

[the remark this answers is not in the document]
Comment (goliney) [reply app/entry.server.jsx:16]: Going through the entire project and replacing spaces with tabs is excessive, IMO. You can use tabs while coding, but before pushing the branch you can <kbd>CMD+R</kbd> and replace tabs with spaces. 

Also, you can use `prettier`. I wish I had set it up before as a part of the project, but you can run it as an external tool.

File: app/entry.server.jsx
[the remark this answers is not in the document]
Comment (goliney) [reply app/entry.server.jsx:1]: These are all the changes in this file:
image

Only spaces and console.log.

To see diff like this you need to hide whitespaces:
image

File: app/services/GetProduct.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import db from '../db.server';
+
+async function getProduct(id) {
```
Comment (goliney) [inline app/services/GetProduct.js:3]: I don't see where this function is used. Could you please point me to the right place?

File: extensions/donate-button/blocks/donate_button.liquid [hunk 5]
```diff
@@ -0,0 +1,97 @@
+
+<div class="donate_wrapper"></div>
+
+<script>
+var product_id = {{product.id}};
+var tunel_url = 'https://norton-barriers-picture-surgeons.trycloudflare.com';
```
[the remark this answers is not in the document]
Comment (goliney) [reply extensions/donate-button/blocks/donate_button.liquid:6]: I replaced the value with the one I found in the generated `toml` config, but the button didn't appear 🤷 

image

**Pull request merged**